### PR TITLE
line 51 : Fix try/catch by adding throw ()

### PR DESCRIPTION
**Pull request by alamenai**
Why catch was empty ?

### CODE REVIEW

**Comment by getify** [comment]
The `catch` is empty because I'm ignoring the error, since it's almost always an error about trying to create a directory that already exists.

It the directory fails to be created (for permissions reasons, etc), the later write to the file (which assumes that directory location) will also fail, which would surface the issue.

**Comment by MenaiAla** [comment]
I'm learning from you @getify. .As far as I know, it's a bad idea to create  `try/catch` if you will not handle the error, because you are silently swallowing an error condition and then continuing execution. Why not using `if` instead. [Ned Batchelder](https://stackoverflow.com/users/14343/ned-batchelder) answer's on this [issue](https://stackoverflow.com/questions/1234343/why-are-empty-catch-blocks-a-bad-idea#:~:text=Generally%2C%20you%20should%20only%20catch,always%20a%20very%20bad%20idea.).

**Comment by getify** [comment]
> it's a bad idea to create try/catch if you will not handle the error

I think the advice may often be valid, but it's rare that such absolute claims are always valid.

In this case, I don't control the fact that `mkdir(..)` throws on a duplicate directory creation. The alternatives are to check to see if the directory exists and only try to create if not. That's valid, but now that's two calls instead of one, and also has a (very slight) race condition. Or I could parse the thrown error to see if it's a duplicate directory error vs some other error. But that's more intricate/brittle than I prefer.

So, in other words, there's a variety of judgement calls to be made here, but as a simple choice made, this is the one I went with.
